src/libraries/System.Private.CoreLib/src/System/Reflection/RuntimeAssemblyName.cs

```diff
@@ -14,9 +14,15 @@ namespace System.Reflection
     //
     // We use this as our internal interchange type and only convert to and from the public AssemblyName class at public boundaries.
     //
-    public sealed class RuntimeAssemblyName : IEquatable<RuntimeAssemblyName>
+#if CORERT
+    [System.Runtime.CompilerServices.ReflectionBlocked]
+    public // Needs to be public so that Reflection.Core can see it.
+#else
+    internal
+#endif
+    sealed class RuntimeAssemblyName : IEquatable<RuntimeAssemblyName>
```

Comment:
Is RuntimeAssemblyName really needed? It looks like unnecessary abstraction. CoreRT reflection has a lot of unnecessary abstractions.

Reply:
`RuntimeAssemblyName` may be useful. It is basically an immutable and lightweight version of AssemblyName. In the native code we have `AssemblySpec`, or more precisely `BaseAssemblySpec`. It is very similar to `RuntimeAssemblyName` and serves a similar purpose. I see how `RuntimeAssemblyName` is useful in CoreRT and as more code in this area is moved to managed, the unified implementation will likely need something like this.

I was more thinking along the lines of eventually making `RuntimeAssemblyName` a struct and use it as internal assembly name "currency". Maybe also unify its layout with the native `BaseAssemblySpec` and use when talking to native code, so that converting from to `AssemblyName` is simpler.

Reply:
> It is basically an immutable and lightweight version of AssemblyName

`AssemblyName`s are rarely instantiated types in a typical program. I doubt that the `RuntimeAssemblyName` code will pay for the few extra `AssemblyName` fields in a typical program.

> Maybe also unify its layout with the native BaseAssemblySpec

I think you would have a problem with strings. Managed types want to have managed strings, unmanaged types want to have unmanaged strings, so you have to convert.

Reply:
Yes, managed strings would be an issue. Byte arrays too. Some conversion would still be needed. But then we could continue using AssemblyName too. 
I thought that we might need a better internal representation for AssemblyName. In particular the part that it can be changed bothers me. If we keep them, like keys in a cache, for example, we would need to make sure they are not shared with the user. It felt like we could use a different and perhaps more lightweight representation, but `RuntimeAssemblyName` may not be that.

I'll move `RuntimeAssemblyName` back to be a CoreRT thing. I think for this change the parser could just produce AssemblyName.